Content.Server/Radio/EntitySystems/MessagesServerSystem.cs

```diff
@@ -0,0 +1,129 @@
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<MessagesServerComponent, DeviceNetworkPacketEvent>(OnPacketReceived);
+    }
+
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+        var serverQuery = EntityQueryEnumerator<MessagesServerComponent>();
+        while (serverQuery.MoveNext(out var uid, out var server))
+        {
+            if (!_singletonServerSystem.IsActiveServer(uid))
+                continue;
+            if (server.NextUpdate <= _gameTiming.CurTime)
+            {
+                server.NextUpdate += server.UpdateDelay;
+
+                Update(uid, server);
+            }
+        }
+    }
+
+    public void Update(EntityUid uid, MessagesServerComponent component)
+    {
+        if (!this.IsPowered(uid, EntityManager))
+            return;
+
+        var serverQuery = EntityManager.AllEntityQueryEnumerator<MessagesServerComponent>();
+
+        while (serverQuery.MoveNext(out var serverUid, out var serverComponent))
+        {
+            component.Messages = new List<MessagesMessageData>(component.Messages.Union(serverComponent.Messages));
+            serverComponent.Messages = new List<MessagesMessageData>(component.Messages);
+        }
+
+        var packet = new NetworkPayload()
+        {
+            ["NameQuery"] = true,
+            ["ServerComponent"] = component
+        };
+        _deviceNetworkSystem.QueuePacket(uid, null, packet);
+    }
+
+    /// <summary>
+    /// Reacts to packets received from clients
+    /// </summary>
+    private void OnPacketReceived(EntityUid uid, MessagesServerComponent component, DeviceNetworkPacketEvent args)
+    {
+        if (!_singletonServerSystem.IsActiveServer(uid))
+            return;
+        if (args.Data.TryGetValue<string>("NewName", out var name) && args.Data.TryGetValue<int>("UserId", out var userId))
+            component.NameDict[userId] = name;
+        if (args.Data.TryGetValue<MessagesMessageData>("Message", out var message))
+            SendMessage(uid, component, message);
+    }
+
+    /// <summary>
+    /// Broadcasts a message into the network
+    /// </summary>
+    public void SendMessage(EntityUid uid, MessagesServerComponent component, MessagesMessageData message)
+    {
+        component.Messages.Add(message);
+
+        var packet = new NetworkPayload()
+        {
+            ["Message"] = message,
+            ["ServerComponent"] = component
+        };
+
+        _deviceNetworkSystem.QueuePacket(uid, null, packet);
+    }
+
+    /// <summary>
+    /// Returns the name of a given user
+    /// </summary>
+    public string GetNameFromDict(MessagesServerComponent component, int key)
+    {
+        if (component.NameDict.TryGetValue(key, out var value))
+            return value;
+        return Loc.GetString("messages-pda-user-missing");
+    }
```

Comment:
This should return a nullable string or be a tryget with a bool return because callers have no idea this failed.